### ISSUE TITLE
Docs: When I test under step seven "Fetching data for the dashboard overview page" I get the error "Error: Failed to fetch revenue data."

### ISSUE THREAD

**Issue by fatih-keon** [vercel/next-learn#947]
### The error comes from the fetchRevenue() function located in the app/lib/data.ts file.

```
export async function fetchRevenue() {
  try {
    // Artificially delay a response for demo purposes.
    // Don't do this in production :)

    // console.log('Fetching revenue data...');
    // await new Promise((resolve) => setTimeout(resolve, 3000));

    const data = await sql<Revenue>`SELECT * FROM revenue`;

    //console.log('Data fetch completed after 3 seconds.');

    return data.rows;
  } catch (error) {
    console.error('Database Error:', error);
    throw new Error('Failed to fetch revenue data.');
  }
}
```
**For the solution, I wanted to use the command we used in the database test before.**

```
import { db } from "@vercel/postgres";

 const client = await db.connect();

 async function listInvoices() {
 	const data = await client.sql`
     SELECT invoices.amount, customers.name
     FROM invoices
     JOIN customers ON invoices.customer_id = customers.id
     WHERE invoices.amount = 666;
   `;

 	return data.rows;
 }

export async function GET() {
   try {
   	return Response.json(await listInvoices());
   } catch (error) {
   	return Response.json({ error }, { status: 500 });
   }
}
```
**And with my novice coding knowledge I wrote this and ran the graph but I didn't understand why the library(import { sql } from '@vercel/postgres';
) wasn't working**

```
import { db } from "@vercel/postgres";

const testr = await db.connect();
export async function fetchRevenue() {
  try {
    const data = await testr.sql<Revenue>`
  SELECT * FROM revenue
  `;
    return data.rows;
  } catch (error) {
    console.error('database error', error);
    throw new Error('revenue datasını çekemedi')
  }
}
```

### Is there any context that might help us understand?

I think I'm missing something

### Does the docs page already exist? Please link to it.

https://nextjs.org/learn/dashboard-app/fetching-data

**Comment by EverDegen** [comment]
import { db } from "@vercel/postgres";


export async function fetchRevenue() {
  try {
const testr = await db.connect(); // move this here to create a connection
    const data = await testr.sql<Revenue>`
  SELECT * FROM revenue
  `;
testr.release(); //add this to release the connection
    return data.rows;
  } catch (error) {
    console.error('database error', error);
    throw new Error('revenue datasını çekemedi')
  }
}

do this for every function and you will be good to go.

**Comment by VitaliJud** [comment]
> Fixed it. Used Supabase DB and that was the issue. Created a new Neon DB and everything works smoothly. There is an announcement here that Posgres is moving to Neon.

@123filip123 Other than installing Neon DB and grabing the variables to update env.local, what other changes did you make? I had a suspicion that the error I'm getting is due to the DB since there was no Postgres access, only Supabase and I did install Neon per your suggestion, but it's missing other variables too, getting error with Neon:
`VercelPostgresError: VercelPostgresError - 'missing_connection_string': You did not supply a 'connectionString' and no 'POSTGRES_URL' env var was found.`

**Comment by fatih-keon** [comment]
> Düzeltildi. Supabase DB kullanıldı ve sorun buydu. Yeni bir Neon DB oluşturuldu ve her şey sorunsuz çalışıyor. Posgres'in Neon'a taşındığına dair burada bir duyuru var.

Yes, that's true, since I opened the db in supabase, there were constant disconnections, I should have read it more carefully.

**Comment by realfr** [comment]
> Fixed it. Used Supabase DB and that was the issue. Created a new Neon DB and everything works smoothly. There is an announcement here that Posgres is moving to Neon.

this worked for me as well. Only changed the `.env` contents with the new snippets then did seeding again.

**Comment by smokinjoe** [comment]
@VitaliJud did you have to visit localhost:3000/seed ?

I'm not getting any fetch errors but I'm seeing zero data in my chart

**Comment by aidanpatrickkane** [comment]
> import { db } from "@vercel/postgres";
> 
> export async function fetchRevenue() { try { const testr = await db.connect(); // move this here to create a connection const data = await testr.sql`SELECT * FROM revenue`; testr.release(); //add this to release the connection return data.rows; } catch (error) { console.error('database error', error); throw new Error('revenue datasını çekemedi') } }
> 
> do this for every function and you will be good to go.

Thank you for this! I was using Supabase, and this fixed it.

**Comment by stuartabrown** [comment]
> import { db } from "@vercel/postgres";
> 
> export async function fetchRevenue() { try { const testr = await db.connect(); // move this here to create a connection const data = await testr.sql`SELECT * FROM revenue`; testr.release(); //add this to release the connection return data.rows; } catch (error) { console.error('database error', error); throw new Error('revenue datasını çekemedi') } }
> 
> do this for every function and you will be good to go.

I'm also following the tutorial and chose Supabase and had the same issue.  This fix worked for me too - many thanks!
